types/vote_test.go

```diff
@@ -68,32 +68,32 @@ func TestVoteSignBytesTestVectors(t *testing.T) {
 		0: {
 			"", &Vote{},
 			// NOTE: Height and Round are skipped here. This case needs to be considered while parsing.
-			[]byte{0x2a, 0x2, 0x12, 0x0, 0x32, 0xb, 0x8, 0x80, 0x92, 0xb8,
-				0xc3, 0x98, 0xfe, 0xff, 0xff, 0xff, 0x1},
+			[]byte{0x22, 0x2, 0x12, 0x0, 0x2a, 0xb, 0x8, 0x80, 0x92, 0xb8, 0xc3, 0x98, 0xfe, 0xff, 0xff, 0xff, 0x1},
 		},
 		// with proper (fixed size) height and round (PreCommit):
 		1: {
 			"", &Vote{Height: 1, Round: 1, Type: tmproto.PrecommitType},
-			[]byte{0x8, 0x2, 0x10, 0x1, 0x18, 0x1, 0x2a, 0x2, 0x12, 0x0, 0x32,
-				0xb, 0x8, 0x80, 0x92, 0xb8, 0xc3, 0x98, 0xfe, 0xff, 0xff, 0xff, 0x1},
+			[]byte{0x8, 0x2, 0x11, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x19, 0x1, 0x0, 0x0,
+				0x0, 0x0, 0x0, 0x0, 0x0, 0x22, 0x2, 0x12, 0x0, 0x2a, 0xb, 0x8, 0x80, 0x92, 0xb8,
+				0xc3, 0x98, 0xfe, 0xff, 0xff, 0xff, 0x1},
```

Comment:
These should now match the old test vectors in 0.33 before the proto change (minus the length prefix).
E.g. https://github.com/tendermint/tendermint/pull/4985/files#diff-a12e7b37187cbe705b897de428e7cec9L75-

Reply:
just tested and they do not. SignedMsgType use to be:
```
type SignedMsgType byte

const (
	// Votes
	PrevoteType   SignedMsgType = 0x01
	PrecommitType SignedMsgType = 0x02

	// Proposals
	ProposalType SignedMsgType = 0x20
)
```

while now it is a proto enum. I moved it to a enum so when the proto files are used in a different language it can easily be used to replicate the same encoding.